### PR TITLE
kdeconnect: choose the first avilable device

### LINKED ISSUE #1612: kdeconnect: choose first available device
When I don't specify a `device_id` in the kdeconnect block it doesn't choose the first device paired and reachable. 

`kdeconnect-cli --list-available` shows one device. When I use the id from there and set it as `device_id` it works.

`kdeconnect-cli --list-devices` shows 7 paired devices but only the 5. device is paired and reachable

### PR DESCRIPTION
**Pull request by omkar-mohanty**
Fixes #1612

### CODE REVIEW

**Comment by omkar-mohanty** [comment]
@MaxVerevkin is this the correct direction to go for? I am a bit unsure if all the cases are handles.

File: src/blocks/kdeconnect.rs [hunk 1]
```diff
@@ -84,14 +84,32 @@ pub async fn run(config: Config, mut api: CommonApi) -> Result<()> {
...
+                    api.hide().await?;
+                    continue;
+                }
+            };
+
+            let (tx_new, rx_new) = mpsc::channel(8);
+
+            rx = rx_new;
+            
+            device = Device::new(&dbus_conn, tx_new, &id).await?;
```
Comment (MaxVerevkin) [inline src/blocks/kdeconnect.rs:109]: I haven't tested this, but it looks like a task with is spawned in `Device::new()` will panic after `rx = rx_new`, which drops `rx`.

**Comment by MaxVerevkin** [comment]
> @MaxVerevkin is this the correct direction to go for? I am a bit unsure if all the cases are handles.

I think a better way to organize things is to do something similar to `bluetooth.rs` and `battery/upower.rs`.

**Comment by omkar-mohanty** [comment]
@MaxVerevkin I organized `kdeconnect.rs` like `bluetooth.rs`, I think it should be ready now.

**Comment by MaxVerevkin** [comment]
When device is not found, the block should wait until a new device is added, then it can try searching again. Right now there is a hot loop...

---

And to be honest, I don't quite understand why I decided to use a tokio task + channels to watch for updates when I was porting this block. It doesn't make sense. Instead a `Device` can contain three streams which are `.next().await`ed in `wait_for_change`.

**Comment by omkar-mohanty** [comment]
@MaxVerevkin for some reason `cargo fmt` does not seem to work in my local machine

**Comment by MaxVerevkin** [comment]
> @MaxVerevkin for some reason `cargo fmt` does not seem to work in my local machine

`cargo fmt` unfortunately does not format files which are included via macros (in our case the `define_blocks!` macro in `blocks.rs`), so if you don't use `rust_analyzer`, you have to run `rustfmt --edition 2021 <file>`.

**Comment by omkar-mohanty** [comment]
@MaxVerevkin is the PR close to what you have in mind?

**Comment by MaxVerevkin** [comment]
@omkar-mohanty CPU usage is still 100%

**Comment by MaxVerevkin** [comment]
Now CPU usage is 100% even if the device is available

**Pull request closed without merging**